31_HLSLPathTracer/app_resources/hlsl/render.comp.hlsl

```diff
@@ -0,0 +1,226 @@
+#include "nbl/builtin/hlsl/cpp_compat.hlsl"
+#include "nbl/builtin/hlsl/glsl_compat/core.hlsl"
+#include "nbl/builtin/hlsl/random/pcg.hlsl"
+#include "nbl/builtin/hlsl/random/xoroshiro.hlsl"
+#ifdef PERSISTENT_WORKGROUPS
+#include "nbl/builtin/hlsl/math/morton.hlsl"
+#endif
+
+#include "nbl/builtin/hlsl/bxdf/reflection.hlsl"
+#include "nbl/builtin/hlsl/bxdf/transmission.hlsl"
+
+// add these defines (one at a time) using -D argument to dxc
+// #define SPHERE_LIGHT
+// #define TRIANGLE_LIGHT
+// #define RECTANGLE_LIGHT
+
+#ifdef SPHERE_LIGHT
+#define SPHERE_COUNT 9
+#define TRIANGLE_COUNT 0
+#define RECTANGLE_COUNT 0
+#endif
+
+#ifdef TRIANGLE_LIGHT
+#define TRIANGLE_COUNT 1
+#define SPHERE_COUNT 8
+#define RECTANGLE_COUNT 0
+#endif
+
+#ifdef RECTANGLE_LIGHT
+#define RECTANGLE_COUNT 1
+#define SPHERE_COUNT 8
+#define TRIANGLE_COUNT 0
+#endif
+
+#define LIGHT_COUNT 1
+#define BXDF_COUNT 7
+
+#include "render_common.hlsl"
+#include "pathtracer.hlsl"
+
+using namespace nbl;
+using namespace hlsl;
+
+NBL_CONSTEXPR uint32_t WorkgroupSize = 512;
```

Comment:
optimal is 64 due to not holding up everything with divergence (lots of small WGs in the same SM co-reisdent)

btw this needs to be in shared header